File: PClassic/2022s/Stubs/12/terrifyingTetra.cpp
#include <iostream>
#include <string>
#include <map>
#include <sstream>
#include <vector>
#include <set>
#include <math.h>
#include <iomanip>

using namespace std;

vector<double> terrifyingTetra(vector<vector<double>> ps) {
    vector<double> centroid;
    return centroid;
}

int main() {
    int t;
    for (cin >> t; t > 0; t--) {
        vector<vector<double>> points;
        for (int j = 0; j < 4; j++) {
            string point0, point1, point2;
            cin >> point0;
            cin >> point1;
            cin >> point2;
            vector<double> temp;
            temp.push_back(stod(point0));
            temp.push_back(stod(point1));
            temp.push_back(stod(point2));
            points.push_back(temp);
        }
        vector<double> output = terrifyingTetra(points);
        cout << setprecision(1) << fixed;
        for (int j = 0; j < output.size(); j++) {
            cout << (round(output[j] * 10.0) / 10.0);
            if (j < output.size() - 1) {
                cout << " ";
            }
        }
		cout << "\n";
    }
}

File: PClassic/2022s/Stubs/12/terrifyingTetra.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.*;
import java.io.*;

public class Main {
    public static double[] terrifyingTetra(double[][] ps) {
        double[] pt = new double[3];
	  return pt;
    }

    public static void main(String[] args) throws Exception {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.valueOf(reader.readLine());
        for (int i = 0; i < t; i++) {
            double[][] points = new double[4][3];
            for (int j = 0; j < 4; j++) {
                String point = reader.readLine();
                String[] split = point.split(" ");
                for (int a = 0; a < 3; a++) {
                    points[j][a] = Double.valueOf(split[a]);
                }
            }
            double[] output = terrifyingTetra(points);
            for (int j = 0; j < output.length; j++) {
                System.out.print((Math.round(output[j] * 10.0) / 10.0) + " ");
            }
            System.out.println();
        }
    }
}